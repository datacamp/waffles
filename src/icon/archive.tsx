import Icon from './icon-base';

type ArchiveProps = Omit<React.ComponentProps<typeof Icon>, 'children'>;

function Archive({ size, ...restProps }: ArchiveProps) {
  return (
    <Icon viewBox="0 0 18 18" size={size} {...restProps}>
      <path
        fill="currentColor"
        d="M8.042 10.581 8.04 1a1 1 0 1 1 2 0l.002 9.593 2.296-2.287a.998.998 0 0 1 1.413.003c.39.392.388 1.025-.003 1.415l-3.949 3.932a.998.998 0 0 1-1.526-.014L4.335 9.708a1 1 0 1 1 1.416-1.415l2.29 2.288zM13.572 6a1 1 0 1 1 0-2H17a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1H1a1 1 0 0 1-1-1V5a1 1 0 0 1 1-1h3.41a1 1 0 1 1 0 2H2v10h14V6h-2.428z"
      />
    </Icon>
  );
}

export default Archive;
