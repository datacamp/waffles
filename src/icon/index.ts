export { default as AddCircleInverted } from './output/add-circle-inverted';
export { default as AddCircle } from './output/add-circle';
export { default as AddUser } from './output/add-user';
export { default as AppleBrand } from './output/apple-brand';
export { default as Apply } from './output/apply';
export { default as Archive } from './output/archive';
export { default as ArrowDown } from './output/arrow-down';
export { default as ArrowLeft } from './output/arrow-left';
export { default as ArrowRight } from './output/arrow-right';
export { default as ArrowUp } from './output/arrow-up';
export { default as Assess } from './output/assess';
export { default as AssignmentInverted } from './output/assignment-inverted';
export { default as Assignment } from './output/assignment';
export { default as Attachment } from './output/attachment';
export { default as AttentionCircleInverted } from './output/attention-circle-inverted';
export { default as AttentionCircle } from './output/attention-circle';
export { default as Back } from './output/back';
export { default as BarChart } from './output/bar-chart';
export { default as Bell } from './output/bell';
export { default as Book } from './output/book';
export { default as BookmarkInverted } from './output/bookmark-inverted';
export { default as Bookmark } from './output/bookmark';
export { default as Branch } from './output/branch';
export { default as Briefcase } from './output/briefcase';
export { default as Building } from './output/building';
export { default as Bulb } from './output/bulb';
export { default as Bullet } from './output/bullet';
export { default as Calendar } from './output/calendar';
export { default as Camera } from './output/camera';
export { default as ChatInverted } from './output/chat-inverted';
export { default as Chat } from './output/chat';
export { default as CheckmarkCircleInverted } from './output/checkmark-circle-inverted';
export { default as CheckmarkCircle } from './output/checkmark-circle';
export { default as Checkmark } from './output/checkmark';
export { default as ChevronDown } from './output/chevron-down';
export { default as ChevronLeft } from './output/chevron-left';
export { default as ChevronRight } from './output/chevron-right';
export { default as ChevronUp } from './output/chevron-up';
export { default as CircleInverted } from './output/circle-inverted';
export { default as Circle } from './output/circle';
export { default as ClockInverted } from './output/clock-inverted';
export { default as Clock } from './output/clock';
export { default as Code } from './output/code';
export { default as CogInverted } from './output/cog-inverted';
export { default as Cog } from './output/cog';
export { default as Columns } from './output/columns';
export { default as Command } from './output/command';
export { default as Compass } from './output/compass';
export { default as Copy } from './output/copy';
export { default as CreditCard } from './output/credit-card';
export { default as CrossCircleInverted } from './output/cross-circle-inverted';
export { default as CrossCircle } from './output/cross-circle';
export { default as Cross } from './output/cross';
export { default as Crosshair } from './output/crosshair';
export { default as DataCampBrand } from './output/data-camp-brand';
export { default as Database } from './output/database';
export { default as Desktop } from './output/desktop';
export { default as Diamond } from './output/diamond';
export { default as Document } from './output/document';
export { default as Dollar } from './output/dollar';
export { default as Download } from './output/download';
export { default as Edit } from './output/edit';
export { default as Education } from './output/education';
export { default as ExerciseInverted } from './output/exercise-inverted';
export { default as Exercise } from './output/exercise';
export { default as Exit } from './output/exit';
export { default as Expand } from './output/expand';
export { default as ExternalLink } from './output/external-link';
export { default as FacebookBrand } from './output/facebook-brand';
export { default as Feed } from './output/feed';
export { default as Filter } from './output/filter';
export { default as Flag } from './output/flag';
export { default as FlashInverted } from './output/flash-inverted';
export { default as Flash } from './output/flash';
export { default as Flashcard } from './output/flashcard';
export { default as Float } from './output/float';
export { default as Folder } from './output/folder';
export { default as Footer } from './output/footer';
export { default as Forbidden } from './output/forbidden';
export { default as Forward } from './output/forward';
export { default as FullScreen } from './output/full-screen';
export { default as GithubBrand } from './output/github-brand';
export { default as Globe } from './output/globe';
export { default as GoogleBrand } from './output/google-brand';
export { default as GoogleColoredBrand } from './output/google-colored-brand';
export { default as GooglePlayBrand } from './output/google-play-brand';
export { default as GroupInverted } from './output/group-inverted';
export { default as Group } from './output/group';
export { default as Header } from './output/header';
export { default as HealthInverted } from './output/health-inverted';
export { default as Health } from './output/health';
export { default as HeartInverted } from './output/heart-inverted';
export { default as Heart } from './output/heart';
export { default as HelpCircleInverted } from './output/help-circle-inverted';
export { default as HelpCircle } from './output/help-circle';
export { default as Help } from './output/help';
export { default as HiddenInverted } from './output/hidden-inverted';
export { default as Hidden } from './output/hidden';
export { default as Home } from './output/home';
export { default as Image } from './output/image';
export { default as InfoCircleInverted } from './output/info-circle-inverted';
export { default as InfoCircle } from './output/info-circle';
export { default as InstagramBrand } from './output/instagram-brand';
export { default as Integration } from './output/integration';
export { default as Interface } from './output/interface';
export { default as Introduction } from './output/introduction';
export { default as KaggleBrand } from './output/kaggle-brand';
export { default as Keyboard } from './output/keyboard';
export { default as Lab } from './output/lab';
export { default as LeanpubBrand } from './output/leanpub-brand';
export { default as Learn } from './output/learn';
export { default as LineChart } from './output/line-chart';
export { default as Link } from './output/link';
export { default as LinkedInBrand } from './output/linked-in-brand';
export { default as List } from './output/list';
export { default as LockedInverted } from './output/locked-inverted';
export { default as Locked } from './output/locked';
export { default as Mail } from './output/mail';
export { default as Markdown } from './output/markdown';
export { default as Matrix } from './output/matrix';
export { default as Medal } from './output/medal';
export { default as Menu } from './output/menu';
export { default as MicrophoneInverted } from './output/microphone-inverted';
export { default as Microphone } from './output/microphone';
export { default as Mobile } from './output/mobile';
export { default as MoonInverted } from './output/moon-inverted';
export { default as Moon } from './output/moon';
export { default as More } from './output/more';
export { default as MuteInverted } from './output/mute-inverted';
export { default as Mute } from './output/mute';
export { default as NineDots } from './output/nine-dots';
export { default as Pause } from './output/pause';
export { default as PaypalBrand } from './output/paypal-brand';
export { default as PaypalColoredBrand } from './output/paypal-colored-brand';
export { default as Phone } from './output/phone';
export { default as PieChart } from './output/pie-chart';
export { default as PinInverted } from './output/pin-inverted';
export { default as Pin } from './output/pin';
export { default as PlatformSwitch } from './output/platform-switch';
export { default as PlayInverted } from './output/play-inverted';
export { default as Play } from './output/play';
export { default as Plus } from './output/plus';
export { default as Practice } from './output/practice';
export { default as Publish } from './output/publish';
export { default as PythonBrand } from './output/python-brand';
export { default as RBrand } from './output/r-brand';
export { default as Random } from './output/random';
export { default as Redo } from './output/redo';
export { default as Refresh } from './output/refresh';
export { default as RemoveInverted } from './output/remove-inverted';
export { default as Remove } from './output/remove';
export { default as Reply } from './output/reply';
export { default as RewindInverted } from './output/rewind-inverted';
export { default as Rewind } from './output/rewind';
export { default as RocketInverted } from './output/rocket-inverted';
export { default as Rocket } from './output/rocket';
export { default as Rows } from './output/rows';
export { default as Save } from './output/save';
export { default as ScalaBrand } from './output/scala-brand';
export { default as Search } from './output/search';
export { default as Share } from './output/share';
export { default as SkipInverted } from './output/skip-inverted';
export { default as Skip } from './output/skip';
export { default as SlackBrand } from './output/slack-brand';
export { default as SlackColoredBrand } from './output/slack-colored-brand';
export { default as SoundInverted } from './output/sound-inverted';
export { default as Sound } from './output/sound';
export { default as SparklesInverted } from './output/sparkles-inverted';
export { default as Sparkles } from './output/sparkles';
export { default as StarInverted } from './output/star-inverted';
export { default as Star } from './output/star';
export { default as StopInverted } from './output/stop-inverted';
export { default as Stop } from './output/stop';
export { default as SunInverted } from './output/sun-inverted';
export { default as Sun } from './output/sun';
export { default as Tab } from './output/tab';
export { default as Tag } from './output/tag';
export { default as Team } from './output/team';
export { default as TerminalInverted } from './output/terminal-inverted';
export { default as Terminal } from './output/terminal';
export { default as Text } from './output/text';
export { default as TrashInverted } from './output/trash-inverted';
export { default as Trash } from './output/trash';
export { default as TrimBegin } from './output/trim-begin';
export { default as TrimEnd } from './output/trim-end';
export { default as Trophy } from './output/trophy';
export { default as TwitterBrand } from './output/twitter-brand';
export { default as Undo } from './output/undo';
export { default as Unlink } from './output/unlink';
export { default as UnlockedInverted } from './output/unlocked-inverted';
export { default as Unlocked } from './output/unlocked';
export { default as Upload } from './output/upload';
export { default as UpvoteInverted } from './output/upvote-inverted';
export { default as Upvote } from './output/upvote';
export { default as UserInverted } from './output/user-inverted';
export { default as User } from './output/user';
export { default as VerifiedInverted } from './output/verified-inverted';
export { default as Verified } from './output/verified';
export { default as VideoCameraInverted } from './output/video-camera-inverted';
export { default as VideoCamera } from './output/video-camera';
export { default as VideoInverted } from './output/video-inverted';
export { default as Video } from './output/video';
export { default as VisibleInverted } from './output/visible-inverted';
export { default as Visible } from './output/visible';
export { default as Waypoint } from './output/waypoint';
export { default as YoutubeBrand } from './output/youtube-brand';
