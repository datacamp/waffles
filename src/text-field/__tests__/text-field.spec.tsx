import { useRef, useEffect } from 'react';
import { render } from '@testing-library/react';

import { AddCircle, ChevronRight } from '../../icon';
import { TextField } from '../index';

const MOCKED_ID = '123abC';

jest.mock('nanoid', () => {
  return {
    nanoid: () => MOCKED_ID,
  };
});

// Most of the functionalities are identical to what Input component offers

function TestRefTextField() {
  const inputRef = useRef<HTMLInputElement>(null);

  useEffect(() => {
    inputRef.current?.focus();
  }, []);

  return (
    <TextField
      ref={inputRef}
      label="Favorite singer"
      placeholder="Taylor Swift"
    />
  );
}

describe('TextField', () => {
  it('renders a basic text field', () => {
    const { container } = render(<TextField label="Favorite singer" />);

    const input = container.querySelector('input');
    const label = container.querySelector('label');

    expect(input).toBeInTheDocument();
    expect(label).toHaveTextContent('Favorite singer');
  });

  it('renders a text field with description and placeholder', () => {
    const { getByPlaceholderText, getByText } = render(
      <TextField
        label="Favorite singer"
        description="Additional description"
        placeholder="Taylor Swift"
      />,
    );

    const input = getByPlaceholderText('Taylor Swift');
    const description = getByText('Additional description');

    expect(input).toBeInTheDocument();
    expect(description).toBeInTheDocument();
  });

  it('renders additional message when text field is required', () => {
    const { getByText } = render(
      <TextField label="Favorite singer" required />,
    );

    const message = getByText(/required/i);

    expect(message).toBeInTheDocument();
  });

  it('renders error message', () => {
    const { getByText } = render(
      <TextField label="Favorite singer" error="Enter correct singer name." />,
    );

    const error = getByText('Enter correct singer name.');

    expect(error).toBeInTheDocument();
  });

  it('input and label are associated by the same ID', () => {
    const { container } = render(<TextField label="Favorite singer" />);

    const input = container.querySelector('input');
    const label = container.querySelector('label');

    expect(label).toHaveAttribute('for', `form-control-${MOCKED_ID}`);
    expect(input).toHaveAttribute('id', `form-control-${MOCKED_ID}`);
  });

  it('input and error are associated by the same ID', () => {
    const { container, getByText } = render(
      <TextField label="Favorite singer" error="Enter correct singer name." />,
    );

    const input = container.querySelector('input');
    const error = getByText('Enter correct singer name.');

    expect(input).toHaveAttribute(
      'aria-errormessage',
      `form-control-error-${MOCKED_ID}`,
    );
    expect(error).toHaveAttribute('id', `form-control-error-${MOCKED_ID}`);
  });

  it('sets the data attribute on the input', () => {
    const { getByTestId } = render(
      <TextField
        data-testid="favorite-singer-text-field"
        label="Favorite singer"
      />,
    );

    const input = getByTestId('favorite-singer-text-field');

    expect(input).toBeInTheDocument();
  });

  it('renders the disabled input', () => {
    const { container } = render(
      <TextField disabled label="Favorite singer" />,
    );

    const input = container.querySelector('input');

    expect(input).toBeDisabled();
  });

  it('renders left icon', () => {
    const { container } = render(
      <TextField iconLeft={<AddCircle />} label="Favorite singer" />,
    );

    const icon = container.querySelector('svg');

    expect(icon).toBeInTheDocument();
  });

  it('renders right enhancer', () => {
    const { container, getByTestId } = render(
      <TextField
        label="Favorite singer"
        enhancerRight={
          <TextField.Enhancer data-testid="right-enhancer">
            <ChevronRight />
          </TextField.Enhancer>
        }
      />,
    );

    const enhancer = getByTestId('right-enhancer');
    const icon = container.querySelector('svg');

    expect(enhancer).toBeInTheDocument();
    expect(enhancer).toContainElement(icon);
  });

  it('accepts ref and could be focused programmatically', () => {
    const { container } = render(<TestRefTextField />);

    const input = container.querySelector('input');

    expect(input).toHaveFocus();
  });

  it('renders snapshot with label and description', () => {
    const { container } = render(
      <TextField
        label="Favorite singer"
        description="Additional description"
        placeholder="Taylor Swift"
      />,
    );

    const textField = container.firstChild;
    expect(textField).toMatchSnapshot();
  });

  it('renders snapshot of required with error', () => {
    const { container } = render(
      <TextField
        label="Favorite singer"
        error="Enter correct singer name."
        required
      />,
    );

    const textField = container.firstChild;
    expect(textField).toMatchSnapshot();
  });

  it('renders snapshot of inverted', () => {
    const { container } = render(
      <TextField
        inverted
        label="Favorite singer"
        description="Additional description"
      />,
    );

    const textField = container.firstChild;
    expect(textField).toMatchSnapshot();
  });

  it('renders snapshot of search type', () => {
    const { container } = render(<TextField type="search" label="Search" />);

    const input = container.firstChild;
    expect(input).toMatchSnapshot();
  });

  it('renders snapshot of password type', () => {
    const { container } = render(
      <TextField type="password" label="Password" />,
    );

    const input = container.firstChild;
    expect(input).toMatchSnapshot();
  });
});
