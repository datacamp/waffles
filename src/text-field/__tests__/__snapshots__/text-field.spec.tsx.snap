// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders snapshot of inverted 1`] = `
.emotion-0 {
  position: relative;
  width: 100%;
}

.emotion-1 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 1.25;
  font-weight: 800;
  color: #FFFFFF;
  margin-bottom: 8px;
}

.emotion-2 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  line-height: 1.5;
  margin-bottom: 8px;
  color: #FFFFFF;
  font-size: 14px;
  line-height: 1.25;
  margin-top: -4px;
}

.emotion-3 {
  position: relative;
  width: 100%;
  height: 36px;
  opacity: 1;
}

.emotion-3::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -2px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.5;
  border: 1px solid #848C92;
  border-radius: 4px;
  outline: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  cursor: text;
  color: #FFFFFF;
  background-color: #213147;
  padding-left: 12px;
  padding-right: 12px;
}

.emotion-4:disabled {
  cursor: default;
}

.emotion-4:-webkit-autofill {
  box-shadow: 0 0 0 36px white inset;
}

.emotion-4::-webkit-search-decoration,
.emotion-4::-webkit-search-cancel-button,
.emotion-4::-webkit-search-results-button,
.emotion-4::-webkit-search-results-decoration {
  display: none;
}

.emotion-4::-ms-clear,
.emotion-4::-ms-reveal {
  display: none;
  width: 0;
  height: 0;
}

.emotion-4::-webkit-input-placeholder {
  opacity: 1;
  color: #9BA3AB;
}

.emotion-4::-moz-placeholder {
  opacity: 1;
  color: #9BA3AB;
}

.emotion-4:-ms-input-placeholder {
  opacity: 1;
  color: #9BA3AB;
}

.emotion-4::placeholder {
  opacity: 1;
  color: #9BA3AB;
}

<div
  class="emotion-0"
>
  <label
    class="emotion-1"
    for="form-control-123abC"
  >
    Favorite singer
  </label>
  <p
    class="emotion-2"
  >
    Additional description
  </p>
  <div
    class="emotion-3"
  >
    <input
      class="emotion-4"
      id="form-control-123abC"
    />
  </div>
</div>
`;

exports[`TextField renders snapshot of password type 1`] = `
.emotion-0 {
  position: relative;
  width: 100%;
}

.emotion-1 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 1.25;
  font-weight: 800;
  color: #05192D;
  margin-bottom: 8px;
}

.emotion-2 {
  position: relative;
  width: 100%;
  height: 36px;
  opacity: 1;
}

.emotion-2::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -2px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.5;
  border: 1px solid #848C92;
  border-radius: 4px;
  outline: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  cursor: text;
  color: #05192D;
  background-color: #FFFFFF;
  padding-left: 12px;
  padding-right: 36px;
}

.emotion-3:disabled {
  cursor: default;
}

.emotion-3:-webkit-autofill {
  box-shadow: 0 0 0 36px white inset;
}

.emotion-3::-webkit-search-decoration,
.emotion-3::-webkit-search-cancel-button,
.emotion-3::-webkit-search-results-button,
.emotion-3::-webkit-search-results-decoration {
  display: none;
}

.emotion-3::-ms-clear,
.emotion-3::-ms-reveal {
  display: none;
  width: 0;
  height: 0;
}

.emotion-3::-webkit-input-placeholder {
  opacity: 1;
  color: #65707C;
}

.emotion-3::-moz-placeholder {
  opacity: 1;
  color: #65707C;
}

.emotion-3:-ms-input-placeholder {
  opacity: 1;
  color: #65707C;
}

.emotion-3::placeholder {
  opacity: 1;
  color: #65707C;
}

.emotion-4 {
  position: absolute;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  top: 0;
  right: 12px;
  color: #05192D;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: auto;
  cursor: pointer;
  padding: 0;
  border: 0;
  border-radius: 4px;
  padding: 2px;
  margin-right: -2px;
  background-color: transparent;
  color: inherit;
  outline: 0;
}

.emotion-5:disabled {
  cursor: inherit;
}

<div
  class="emotion-0"
>
  <label
    class="emotion-1"
    for="form-control-123abC"
  >
    Password
  </label>
  <div
    class="emotion-2"
  >
    <input
      autocomplete="new-password"
      class="emotion-3"
      id="form-control-123abC"
      spellcheck="false"
      type="password"
    />
    <div
      class="emotion-4"
    >
      <button
        aria-label="Show password text"
        class="emotion-5"
        type="button"
      >
        <svg
          aria-hidden="true"
          height="16"
          viewBox="0 0 18 18"
          width="16"
        >
          <path
            d="M9 14c3.866 0 7-4.367 7-5s-3.134-5-7-5-7 4.102-7 5c0 .898 3.134 5 7 5Zm0 2c-4.209 0-9-4.303-9-7s4.791-7 9-7c4.142 0 9 4.49 9 7s-4.858 7-9 7Zm0-4a3 3 0 1 1 0-6 3 3 0 0 1 0 6Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`TextField renders snapshot of required with error 1`] = `
.emotion-0 {
  position: relative;
  width: 100%;
}

.emotion-1 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 1.25;
  font-weight: 800;
  color: #05192D;
  margin-bottom: 8px;
}

.emotion-2 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  font-size: 12px;
  color: #65707C;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: right;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-3 {
  font-size: 14px;
  color: #DD3400;
}

.emotion-4 {
  position: relative;
  width: 100%;
  height: 36px;
  opacity: 1;
}

.emotion-4::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -2px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.5;
  border: 1px solid #848C92;
  border-radius: 4px;
  outline: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  cursor: text;
  color: #05192D;
  background-color: #FFFFFF;
  padding-left: 12px;
  padding-right: 12px;
  border-color: #DD3400;
  box-shadow: 0 0 0 1px #DD3400;
}

.emotion-5:disabled {
  cursor: default;
}

.emotion-5:-webkit-autofill {
  box-shadow: 0 0 0 36px white inset;
}

.emotion-5::-webkit-search-decoration,
.emotion-5::-webkit-search-cancel-button,
.emotion-5::-webkit-search-results-button,
.emotion-5::-webkit-search-results-decoration {
  display: none;
}

.emotion-5::-ms-clear,
.emotion-5::-ms-reveal {
  display: none;
  width: 0;
  height: 0;
}

.emotion-5::-webkit-input-placeholder {
  opacity: 1;
  color: #65707C;
}

.emotion-5::-moz-placeholder {
  opacity: 1;
  color: #65707C;
}

.emotion-5:-ms-input-placeholder {
  opacity: 1;
  color: #65707C;
}

.emotion-5::placeholder {
  opacity: 1;
  color: #65707C;
}

.emotion-6 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  position: absolute;
  z-index: 1;
  bottom: -8px;
  left: 0;
  -webkit-transform: translateY(100%);
  -moz-transform: translateY(100%);
  -ms-transform: translateY(100%);
  transform: translateY(100%);
  color: #C01100;
  line-height: 1.25;
}

<div
  class="emotion-0"
>
  <label
    class="emotion-1"
    for="form-control-123abC"
  >
    Favorite singer
    <span
      aria-hidden="true"
      class="emotion-2"
    >
      <span
        class="emotion-3"
      >
        â€¢
      </span>
       Required
    </span>
  </label>
  <div
    class="emotion-4"
  >
    <input
      aria-describedby="form-control-error-123abC"
      aria-errormessage="form-control-error-123abC"
      aria-invalid="true"
      class="emotion-5"
      id="form-control-123abC"
      required=""
    />
  </div>
  <p
    class="emotion-6"
    id="form-control-error-123abC"
  >
    Enter correct singer name.
  </p>
</div>
`;

exports[`TextField renders snapshot of search type 1`] = `
.emotion-0 {
  position: relative;
  width: 100%;
}

.emotion-1 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 1.25;
  font-weight: 800;
  color: #05192D;
  margin-bottom: 8px;
}

.emotion-2 {
  position: relative;
  width: 100%;
  height: 36px;
  opacity: 1;
}

.emotion-2::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -2px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-3 {
  position: absolute;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  pointer-events: none;
  top: 0;
  left: 12px;
  color: #05192D;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.5;
  border: 1px solid #848C92;
  border-radius: 4px;
  outline: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  cursor: text;
  color: #05192D;
  background-color: #FFFFFF;
  padding-left: 36px;
  padding-right: 12px;
}

.emotion-4:disabled {
  cursor: default;
}

.emotion-4:-webkit-autofill {
  box-shadow: 0 0 0 36px white inset;
}

.emotion-4::-webkit-search-decoration,
.emotion-4::-webkit-search-cancel-button,
.emotion-4::-webkit-search-results-button,
.emotion-4::-webkit-search-results-decoration {
  display: none;
}

.emotion-4::-ms-clear,
.emotion-4::-ms-reveal {
  display: none;
  width: 0;
  height: 0;
}

.emotion-4::-webkit-input-placeholder {
  opacity: 1;
  color: #65707C;
}

.emotion-4::-moz-placeholder {
  opacity: 1;
  color: #65707C;
}

.emotion-4:-ms-input-placeholder {
  opacity: 1;
  color: #65707C;
}

.emotion-4::placeholder {
  opacity: 1;
  color: #65707C;
}

<div
  class="emotion-0"
>
  <label
    class="emotion-1"
    for="form-control-123abC"
  >
    Search
  </label>
  <div
    class="emotion-2"
  >
    <div
      class="emotion-3"
    >
      <svg
        aria-hidden="true"
        data-testid="search-icon"
        height="16"
        viewBox="0 0 18 18"
        width="16"
      >
        <path
          d="m13.26 12.303 4.408 3.953a1 1 0 0 1-1.336 1.488L11.78 13.66a7.5 7.5 0 1 1 1.482-1.358ZM7.5 13a5.5 5.5 0 1 0 0-11 5.5 5.5 0 0 0 0 11Z"
          fill="currentColor"
        />
      </svg>
    </div>
    <input
      class="emotion-4"
      id="form-control-123abC"
      type="search"
    />
  </div>
</div>
`;

exports[`TextField renders snapshot with label and description 1`] = `
.emotion-0 {
  position: relative;
  width: 100%;
}

.emotion-1 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 1.25;
  font-weight: 800;
  color: #05192D;
  margin-bottom: 8px;
}

.emotion-2 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  line-height: 1.5;
  margin-bottom: 8px;
  color: #05192D;
  font-size: 14px;
  line-height: 1.25;
  margin-top: -4px;
}

.emotion-3 {
  position: relative;
  width: 100%;
  height: 36px;
  opacity: 1;
}

.emotion-3::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -2px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.5;
  border: 1px solid #848C92;
  border-radius: 4px;
  outline: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  cursor: text;
  color: #05192D;
  background-color: #FFFFFF;
  padding-left: 12px;
  padding-right: 12px;
}

.emotion-4:disabled {
  cursor: default;
}

.emotion-4:-webkit-autofill {
  box-shadow: 0 0 0 36px white inset;
}

.emotion-4::-webkit-search-decoration,
.emotion-4::-webkit-search-cancel-button,
.emotion-4::-webkit-search-results-button,
.emotion-4::-webkit-search-results-decoration {
  display: none;
}

.emotion-4::-ms-clear,
.emotion-4::-ms-reveal {
  display: none;
  width: 0;
  height: 0;
}

.emotion-4::-webkit-input-placeholder {
  opacity: 1;
  color: #65707C;
}

.emotion-4::-moz-placeholder {
  opacity: 1;
  color: #65707C;
}

.emotion-4:-ms-input-placeholder {
  opacity: 1;
  color: #65707C;
}

.emotion-4::placeholder {
  opacity: 1;
  color: #65707C;
}

<div
  class="emotion-0"
>
  <label
    class="emotion-1"
    for="form-control-123abC"
  >
    Favorite singer
  </label>
  <p
    class="emotion-2"
  >
    Additional description
  </p>
  <div
    class="emotion-3"
  >
    <input
      class="emotion-4"
      id="form-control-123abC"
      placeholder="Taylor Swift"
    />
  </div>
</div>
`;
